### DISCUSSION TITLE
Faster way to explore objects and Jinja export templating?

### DISCUSSION THREAD

**Discussion by etfz** [netbox-community/netbox#17588] [Help Wanted!] [answered]
Hi,

(Still) trying to write export templates, but the workflow is really rather frustrating. Trying to figure out what properties are available, what types those are, what's the path to such and such property, edit the template in that tiny text area, go to my other tab and refresh the render config, find that my template did not do what I expected (for reasons unknown), go back to the other tab and open the template for editing again, rinse and repeat. On top of that, the web application is not the fastest I've seen. (may be an issue with my server, of course)

`nbshell` has been suggested before, but it does not seem immensely user friendly. What I really want is instant rendered template previewing, as well as some sort of online documentation of objects and their properties. I don't know what's Jinja and what's NetBox.

**Comment by etfz** [comment, marked as answer]
Well, I found https://j2live.ttl255.com, which is a good start. Now I just need a way to export all data that's available to the template.

**Reply by etfz** [reply]
A lot of NetBox (Python?) specific stuff will not be available to the tool anyway, so I'll just make up my own data. This is probably as good as it will get.

**Comment by trakennedy** [comment]
Did you ever find an easier way to explore? I'm in this same situation trying to figure out how to export devices with their IPs and MAC addresses in the same report.

**Reply by etfz** [reply]
Not really, no. I think I've mostly tried figuring things out using the builtin API browser.
